pika/adapters/select_connection.py

```diff
@@ -27,20 +27,15 @@
 ERROR = 0x0008
 
 
-if pika.compat.PY2:
-    _SELECT_ERROR = select.error
-else:
-    # select.error was deprecated and replaced by OSError in python 3.3
-    _SELECT_ERROR = OSError
+_SELECT_ERRORS = (select.error, OSError, IOError)
+_select_error_checkers = {
+    select.error : lambda e: e.args[0]==errno.EINTR,
+    OSError      : lambda e: e.errno  ==errno.EINTR,
+    IOError      : lambda e: e.errno  ==errno.EINTR,
+    }
 
-
-def _get_select_errno(error):
-    if pika.compat.PY2:
-        assert isinstance(error, select.error), repr(error)
-        return error.args[0]
-    else:
-        assert isinstance(error, OSError), repr(error)
-        return error.errno
+def _is_resumable(exc):
```

Comment:
This could benefit from a docstring, taking care to document the arg; something like

```
:param exc: exception; must be of one of the exception classes in _SELECT_ERRORS
```